net: l2: openthread: Fix the network initialization when CONFIG_NET_CONFIG_MY_IPV6_ADDR is not set

lmaciejonczyk opened the pull request:
It has been needed to set CONFIG_NET_CONFIG_MY_IPV6_ADDR to start the OpenThread network without any errors/unnecessary delays from network configuration library so far. The OpenThread provides needed IPv6 addresses therefore manually setting is unnecessary.

jukkar reviewed (commented):
I am not sure why would we need this. When ip address is set to the network interface, net mgmt interface event is/should be sent and things should proceed normally.

rlubos reviewed (approved):
LGTM

rlubos commented:
@jukkar Currently the `NET_CONFIG_NEED_IPV6` is only notified when DAD is used:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/config/init.c#L238

Thraed networks do not use DAD (it has its own mechanism for duplicate detection). Probably we should rework the logic in `init.c` instead to notifiy `NET_CONFIG_NEED_IPV6` immediately if DAD is disabled?

lmaciejonczyk commented:
> @jukkar Currently the `NET_CONFIG_NEED_IPV6` is only notified when DAD is used:
> https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/config/init.c#L238
> 
> Thraed networks do not use DAD (it has its own mechanism for duplicate detection). Probably we should rework the logic in `init.c` instead to notifiy `NET_CONFIG_NEED_IPV6` immediately if DAD is disabled?

@jukkar The solution proposed by @rlubos  is fine for me too. Should I change the approach?

jukkar commented:
> The solution proposed by @rlubos is fine for me too. Should I change the approach?

Sorry for late reply, just got back from vacation. Yes, please rework the PR as suggested by @rlubos

lmaciejonczyk committed:
net: config: Fix IPv6 setup when DAD is unused and IPv6 addr is not set

When the dupplicate address detection mechanism is not used, we should
not wait for IPv6 address configuration. Because of the bug we are
waiting for it, when CONFIG_NET_CONFIG_MY_IVP6_ADDR is not set.
This commit fixes it.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

lmaciejonczyk commented:
> > The solution proposed by @rlubos is fine for me too. Should I change the approach?
> 
> Sorry for late reply, just got back from vacation. Yes, please rework the PR as suggested by @rlubos

Done

rlubos reviewed (approved):
This should work as well
